src/lib/apiTypesV2/Entity.h

```diff
@@ -64,6 +64,7 @@ class Entity
   double                  modDate;          // used by dateModified functionality in NGSIv2
 
   std::vector<ProvidingApplication> providingApplicationList;    // Not part of NGSI, used internally for CPr forwarding functionality
+  std::string                       forwardingMode;              // Not part of NGSI, used internally for CPr forwarding functionality
```

Comment:
This assumes that all providing applications in the list are using the same forwarding mode, but this is not necesarily true as each providing application may come from a different registration, using different modes. It would be better to have `std:.vector<string> forwardingModeList`, so the i-th element in `providingApplicationList` corresponds to the i-th element in `fowardingModeList`.

In addition, I understand this refers to the nature of the CPr, either a legacy one implementing NGSIv1 or not legacy and implementing NGSIv2. In that case, I'd suggest a different name, e.g. `providerTypeList`. Note that the "forwarding mode" is a different thing in the NGSIv2 spec (in particular a way of specifying if CPr is able to process queries, updates of both) and out of the scope of this PR.

Finally, maybe it should a enum instead of string, Currently only two elements in the enum ("v1" and "v2") but in the future it could be expanded if new forwarding mechanism are defined.

Reply:
Looking to how this apperas in the DB:

> **format**: the format to use to send forwarded requests. The only accepted value for now is **JSON** (meaning NGSIv1 format), although this may change in the future.

I change my suggestion to: `providerFormatList`.

Reply:
And enum: `JSON` and `V2`.

Why V2 instead of "normalized"? This is not the same than NGSIv2 representation modes (normalized, keyValues, values) and I think `V2` is enough.

Reply:
Yeah, what I would have wanted would be a 'apiVersion'.
Instead we have to use "JSON" for APIv1 and "normalized"/"keyValues"/"values" or something similar for APIv2. Not perfect.
For now, just to move on I am using ONLY "normalized".

Perhaps I haven't fully understood the providingApplicationList ...
Didn't really stop and consider it ...
If it is as you say, I believe the forwardingMode should be part of the providingApplication

Reply:
Mabye I'm not make me clear enough. I'll try again :)

* "forwaringMode" term is about the kind of operations that de CPr accept (query/updates/both). Out of the scope of this PR.
* "format" (or whichever other name we want to use) is about the version of NGSI that the CPr understands. JSON for NGSIv1 (for legacy reasons) or V2 for NGSIv2 (normalized is also fine).

Please, use the proper name for the varible. You have all the code plentiy of "fowardingMode" but what you really mean is "format" or whichever other name we want to use).

Reply:
With regards to:

> If it is as you say, I believe the ~~forwardingMode~~ format should be part of the providingApplication

It makes sense. Do we have already some struct/class modeling Cprs in which we could base? Or would it be a new one?

Reply:
We have a class called ProvidingApplication.
Haven't looked at it yet, but I hope to be able to put this field inside that class

Reply:
I think that class is intended to model the provider in the "scope" of a context registration request/response. However, it seems pretty appropiate also for this case (vs. the alternative of inventing a new class for so close concept).

Reply:
Fixed in c522e4eaf6d7bb140f7f12c7815049bcd23ff200